kube-core/src/request.rs

```diff
@@ -148,9 +148,33 @@ impl Request {
             qp.append_pair("labelSelector", labels);
         }
         let urlstr = qp.finish();
-        let body = serde_json::to_vec(&dp).map_err(Error::SerializeBody)?;
+
+        let data = if std::env::var("KUBE_UNSTABLE_V1_25_DELETE").is_ok() {
+            // Empty delete params triggers an error in kubernetes v1.25
+            // https://github.com/kubernetes/kubernetes/issues/111985
+
+            // This is a hacky temporary branch that we hope we can remove
+            // in the next version of kube when upstream resolves the issue.
+            #[derive(serde::Serialize)]
+            #[serde(rename_all = "camelCase")]
+            struct KindWrapper<'a> {
+                api_version: &'a str,
+                kind: &'a str,
+                #[serde(flatten)]
+                data: serde_json::Value,
+            }
+            let kind_wrapped_dp = KindWrapper {
+                api_version: "meta.k8s.io/v1",
+                kind: "DeleteOptions",
+                data: serde_json::to_value(&dp).map_err(Error::SerializeBody)?,
+            };
+            serde_json::to_vec(&kind_wrapped_dp).map_err(Error::SerializeBody)?
+        } else {
+            serde_json::to_vec(&dp).map_err(Error::SerializeBody)?
+        };
```

Comment:
This sucks, but I can't think of a way either :(

---

Maybe add a special case for the default `dp` like `k8s_openapi`? 

1. If `dp` is the default, send `vec![]`
2. Otherwise, if the evar is defined, wrap it
3. Otherwise, serialize as is

Reply:
Done in https://github.com/kube-rs/kube-rs/pull/1008/commits/f44041560e350b2276e44ba9632ee3455a3adf31.
This makes `delete_collection` slightly more resilient on 1.25 (but still breaks without the evar when changing delete params).